### ISSUE TITLE
Chat Component: Attached property not honored when reactionGroup is a component or empty array

### ISSUE THREAD

**Issue by johannao76** [microsoft/fluentui#16146]
### Environment Information

- **Package version(s)**: 0.51.3

### Repro
CodePen; https://codesandbox.io/s/fluent-ui-example-forked-j1kui?file=/example.js

In the chat component, when we pass `attached` as true and `reactionGroup` as either an empty array or a component that returns null/undefined, the attached property is not honored at all.

#### Actual behavior:

The chat message is not attached to the previous one

#### Expected behavior:

The chat message should be attached since there are actually no reactions

### Priorities and help requested:

Requested priority: Normal

Products/sites affected: Teams chat

**Comment by johannao76** [comment]
@layershifter I worked around it already but was filing this to get the issue fixed. It is counterintuitive to have the attached prop not work just because the collection of reactions is empty.

**Comment by layershifter** [comment]
> @layershifter I worked around it already but was filing this to get the issue fixed. It is counterintuitive to have the attached prop not work just because the collection of reactions is empty.

Agree, the main issue that is there no intuitive solution for your request 🤔

**Requirement:** We need to make slot always visible if reactions are present, otherwise it will be confusing for user as reactions will be invisible for attached messages
**Current solution:** Perform a boolean check

https://github.com/microsoft/fluentui/blob/aa34c48589728cf0ed77b856ddaee16e8ca5bfbe/packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx#L218

### Other options?

For arrays we can try:

```jsx
<Chat.Message reactionGroup={[]} />;
//
hasReactionGroup: !!reactionGroup || Array.isArray(reactionGroup) && reactionGroup.length > 0,
````

Then we also need to handle a case with `items`:

```jsx
<Chat.Message reactionGroup={{ items: [] }} />;
//
hasReactionGroup: !!reactionGroup || (Array.isArray(reactionGroup) && reactionGroup.length > 0) || Array.isArray(reactionGroup?.items) && reactionGroup?.items.length > 0,
```

We also have a case with `children` function API:

```jsx
<Chat.Message reactionGroup={{ children: () => null }} />;
// 💥 What we can do there?
```

And then even a case that you specified:

```tsx
const ReactionRenderer = () => {
  return null;
};

// 💡  <ReactionRenderer /> is a JSX element will be an object

{ 
  $$typeof: Symbol(react.element)
  key: null
  props: {}
  ref: null
  type: ƒ ReactionRenderer()
  _owner: null
}
```

In this case I don't have any how we can handle this.

---

To summarize, while I agree that it's not intuitive I don't think that there is a solid solution for the whole library. If you have ideas, please let me know.
